clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary {
+  background: var(--fides-consent-banner-button-tertiary-background);
+  color: var(--fides-consent-banner-button-tertiary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary:hover {
+  background: var(--fides-consent-banner-button-tertiary-hover-background);
+}
+`
+
+/**
+ * Builds a 'style' element containing the CSS styles for the consent banner
+ */
+const buildStyles = (): Node => {
+  const style = document.createElement("style");
+  style.innerHTML = globalBannerStyle;
+  return style;
+};
+
+/**
+ * Initialize the Fides Consent Banner, with optional extraOptions to override defaults.
+ * 
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initBanner = async (defaults: CookieKeyConsent, extraOptions?: ConsentBannerOptions): Promise<void> => {
+  // If the user provides any extra options, override the defaults
+  debugLog("Initializing Fides consent banner with consent cookie defaults...", defaults);
+  if (extraOptions !== undefined) {
+    if (typeof extraOptions !== "object") {
+      // eslint-disable-next-line no-console
+      console.error("Invalid 'extraOptions' arg for Fides.banner(), ignoring", extraOptions);
+    } else {
+      setBannerOptions({ ...getBannerOptions(), ...extraOptions });
+    }
+  }
+  const options: ConsentBannerOptions = getBannerOptions();
+
+  debugLog("Validating Fides consent banner options...", options);
+  if (!validateBannerOptions(options)) {
+    return;
+  }
+
+  if (options.isDisabled) {
+    debugLog("Fides consent banner is disabled, skipping banner initialization!");
+    return;
+  }
+
+  if (hasSavedConsentCookie()) {
+    debugLog("Fides consent cookie already exists, skipping banner initialization!");
+    return;
+  }
+
+  debugLog("Fides consent banner is enabled and consent cookie does not exist. Continuing...");
+
+  if (options.isGeolocationEnabled) {
+    debugLog("Fides consent banner geolocation is enabled. Getting user location...");
+    const location = await getLocation();
+
+    debugLog("Checking if Fides consent banner should be displayed for location:", location);
+    if (!isBannerEnabledForLocation(location)) {
+      debugLog("Fides consent banner is not enabled for location, skipping banner initialization!");
+      return;
+    }
+  } else {
+    debugLog("Fides consent banner geolocation is not enabled. Continuing...");
+  }
+
+  debugLog("Fides consent banner should be shown! Building banner elements & styles...");
+  const banner = buildBanner(defaults);
+  const styles = buildStyles();
+
+  debugLog("Adding Fides consent banner CSS & HTML into the DOM...");
+  document.head.appendChild(styles);
+  document.body.appendChild(banner);
+
+  // Show the banner after a small delay, to allow animation to occur
+  await setTimeout(() => showBanner(banner), 100);
```

Comment:
Minor nit, but this should likely all be wrapped into a separate function that is "concerned" with the DOM APIs and the semantics of building the banner and adding to the document, etc.